### PR TITLE
Add remote localization lookup

### PR DESCRIPTION
**Pull request by DDavidPrime**
Addresses ticket: #587

### CODE REVIEW

File: packages/rx_bloc_cli/example/README.md [hunk 1]
```diff
@@ -268,6 +268,8 @@ If there are new keys added to the main translation file they can be propagated
 
 Upon rebuild, your translations are auto-generated inside `lib/assets.dart`. In order to use them, you need to import the `l10n.dart` file from `lib/l10n/l10n.dart` and then access the translations from your BuildContext via `context.l10n.someTranslationKey` or `context.l10n.featureName.someTranslationKey`.
 
+Localization lookups are also supported. That means that you can request any remote localizations from a dedicated translations endpoint at app start. Grabbing any remote localization for existing features will replace the local translations with new ones.
```
Comment (StanevPrime) [inline packages/rx_bloc_cli/example/README.md:271]: Please add documentation regarding the expected json structure either in the main README.md file or in a sperate file.

File: packages/rx_bloc_cli/mason_templates/bricks/feature_counter/__brick__/lib/l10n/network_lookup/counter_lookup.dart [hunk 2]
```diff
@@ -0,0 +1,33 @@
+{{> licence.dart}}
```
Comment (StanevPrime) [inline packages/rx_bloc_cli/mason_templates/bricks/feature_counter/__brick__/lib/l10n/network_lookup/counter_lookup.dart:1]: Please confirm this lookup is added conditionally based on the executable input

File: packages/rx_bloc_cli/mason_templates/bricks/feature_deeplink/__brick__/lib/l10n/network_lookup/deep_link_lookup.dart [hunk 3]
```diff
@@ -0,0 +1,33 @@
+{{> licence.dart}}
```
Comment (StanevPrime) [inline packages/rx_bloc_cli/mason_templates/bricks/feature_deeplink/__brick__/lib/l10n/network_lookup/deep_link_lookup.dart:1]: Same as above

File: packages/rx_bloc_cli/mason_templates/bricks/feature_deeplink/__brick__/lib/l10n/network_lookup/enter_message_lookup.dart [hunk 4]
```diff
@@ -0,0 +1,36 @@
+{{> licence.dart}}
```
Comment (StanevPrime) [inline packages/rx_bloc_cli/mason_templates/bricks/feature_deeplink/__brick__/lib/l10n/network_lookup/enter_message_lookup.dart:1]: Do we really need lookup for the enter message? Isn't it part of the deep link feature ?

File: packages/rx_bloc_cli/mason_templates/bricks/feature_otp/__brick__/lib/l10n/network_lookup/otp_lookup.dart [hunk 5]
```diff
@@ -0,0 +1,33 @@
+{{> licence.dart}}
```
Comment (StanevPrime) [inline packages/rx_bloc_cli/mason_templates/bricks/feature_otp/__brick__/lib/l10n/network_lookup/otp_lookup.dart:1]: Same as above

File: packages/rx_bloc_cli/mason_templates/bricks/feature_widget_toolkit/__brick__/lib/l10n/network_lookup/widget_toolkit_lookup.dart [hunk 6]
```diff
@@ -0,0 +1,36 @@
+{{> licence.dart}}
```
Comment (StanevPrime) [inline packages/rx_bloc_cli/mason_templates/bricks/feature_widget_toolkit/__brick__/lib/l10n/network_lookup/widget_toolkit_lookup.dart:1]: Same as above

File: packages/rx_bloc_cli/mason_templates/bricks/lib_auth_matrix/__brick__/lib/l10n/network_lookup/auth_matrix_lookup.dart [hunk 7]
```diff
@@ -0,0 +1,34 @@
+{{> licence.dart}}
```
Comment (StanevPrime) [inline packages/rx_bloc_cli/mason_templates/bricks/lib_auth_matrix/__brick__/lib/l10n/network_lookup/auth_matrix_lookup.dart:1]: Same as above

File: packages/rx_bloc_cli/mason_templates/bricks/rx_bloc_base/__brick__/README.md [hunk 8]
```diff
@@ -268,6 +268,8 @@ If there are new keys added to the main translation file they can be propagated
 
 Upon rebuild, your translations are auto-generated inside `lib/assets.dart`. In order to use them, you need to import the `l10n.dart` file from `lib/l10n/l10n.dart` and then access the translations from your BuildContext via `context.l10n.someTranslationKey` or `context.l10n.featureName.someTranslationKey`.
 
+Localization lookups are also supported. That means that you can request any remote localizations from a dedicated translations endpoint at app start. Grabbing any remote localization for existing features will replace the local translations with new ones.
```
Comment (StanevPrime) [inline packages/rx_bloc_cli/mason_templates/bricks/rx_bloc_base/__brick__/README.md:271]: Same as above

File: packages/rx_bloc_cli/mason_templates/bricks/rx_bloc_base/__brick__/lib/base/data_sources/local/translations_shared_data_source.dart [hunk 9]
```diff
@@ -0,0 +1,24 @@
+{{> licence.dart }}
+
+import 'dart:convert';
+
+import '../../models/i18n/i18n_models.dart';
+import 'shared_preferences_instance.dart';
+import 'translations_local_data_source.dart';
+
+class TranslationsSharedDataSource extends TranslationsDataSource {
```
Comment (StanevPrime) [inline packages/rx_bloc_cli/mason_templates/bricks/rx_bloc_base/__brick__/lib/base/data_sources/local/translations_shared_data_source.dart:9]: Please remove the local storage providers. Let's use the in-memory storage ot keep the solution as minimal as possible

File: packages/rx_bloc_cli/mason_templates/bricks/rx_bloc_base/__brick__/lib/base/common_services/translations_service.dart [hunk 10]
```diff
@@ -0,0 +1,22 @@
+import '../models/i18n/i18n_models.dart';
```
Comment (StanevPrime) [inline packages/rx_bloc_cli/mason_templates/bricks/rx_bloc_base/__brick__/lib/base/common_services/translations_service.dart:1]: Please move all translation related files in a single `lib` such as lib_translations

File: packages/rx_bloc_cli/mason_templates/bricks/rx_bloc_base/__brick__/lib/base/di/{{project_name}}_with_dependencies.dart [hunk 11]
```diff
@@ -172,6 +177,12 @@ class {{project_name.pascalCase()}}WithDependencies extends StatelessWidget {
             baseUrl: config.baseUrl,
           ),
         ),{{/has_authentication}}
+        Provider<TranslationsDataSource>(
```
Comment (StanevPrime) [inline packages/rx_bloc_cli/mason_templates/bricks/rx_bloc_base/__brick__/lib/base/di/{{project_name}}_with_dependencies.dart:180]: Please create a di for the `lib_translations`

File: packages/rx_bloc_cli/mason_templates/bricks/rx_bloc_base/__brick__/lib/base/models/i18n/i18n_models.dart [hunk 12]
```diff
@@ -0,0 +1,18 @@
+{{> licence.dart }}
+
+import 'package:json_annotation/json_annotation.dart';
+
+part 'i18n_models.g.dart';
+
+@JsonSerializable()
```
Comment (StanevPrime) [inline packages/rx_bloc_cli/mason_templates/bricks/rx_bloc_base/__brick__/lib/base/models/i18n/i18n_models.dart:7]: Please move the models to the lib_translations

File: packages/rx_bloc_cli/mason_templates/bricks/rx_bloc_base/__brick__/lib/base/repositories/translations_repository.dart [hunk 13]
```diff
@@ -0,0 +1,14 @@
+import '../common_mappers/error_mappers/error_mapper.dart';
+import '../data_sources/local/translations_local_data_source.dart';
+import '../models/i18n/i18n_models.dart';
+
+class TranslationsRepository extends TranslationsDataSource {
```
Comment (StanevPrime) [inline packages/rx_bloc_cli/mason_templates/bricks/rx_bloc_base/__brick__/lib/base/repositories/translations_repository.dart:5]: Same as above

**Review by StanevPrime — CHANGES_REQUESTED** [review]

File: packages/rx_bloc_cli/mason_templates/bricks/feature_counter/__brick__/lib/l10n/network_lookup/counter_lookup.dart (hunk 2 above)
Comment (DDavidPrime) [reply packages/rx_bloc_cli/mason_templates/bricks/feature_counter/__brick__/lib/l10n/network_lookup/counter_lookup.dart:1]: Yes, in fact, it is added conditionally. The `Counter` feature is locked behind the `enable-feature-counter` flag and thus won't generate if not specified. This is for this and the other items mentioned. So this should be good unless I'm mistaken for something else.

Comment (StanevPrime) [reply packages/rx_bloc_cli/mason_templates/bricks/feature_counter/__brick__/lib/l10n/network_lookup/counter_lookup.dart:1]: I see. That's great!

File: packages/rx_bloc_cli/example/README.md [hunk 14]
```diff
@@ -268,6 +268,39 @@ If there are new keys added to the main translation file they can be propagated
 
 Upon rebuild, your translations are auto-generated inside `lib/assets.dart`. In order to use them, you need to import the `l10n.dart` file from `lib/l10n/l10n.dart` and then access the translations from your BuildContext via `context.l10n.someTranslationKey` or `context.l10n.featureName.someTranslationKey`.
 
+#### Remote localization lookup
+
+Localization lookups are also supported. That means that you can request any remote localizations from a dedicated translations endpoint at app start. Grabbing any remote localization for existing features will replace the local translations with new ones.
```
Comment (StanevPrime) [inline packages/rx_bloc_cli/example/README.md:273]: Please add information that the localisations are loaded in the splash screen

File: packages/rx_bloc_cli/mason_templates/bricks/rx_bloc_base/__brick__/README.md [hunk 15]
```diff
@@ -268,6 +268,39 @@ If there are new keys added to the main translation file they can be propagated
...
+      },
+      "bg":{
+         "_ok":"ok",
+         "login___logIn":"Вход с имейл"
+      }
+   }
+}
+```
+
+One thing to note is that the keys in the translations follow a naming scheme which consists of three parts: the feature name, separator and the translation key. The feature name by default represents the feature under which the translations are categorized followed by two underscores. It is optional and if omitted, the key overrides are placed within the main translations. The separator is a character used to separate the feature name and the translation key. The translation key is the key within the specified feature used in the app.
```
Comment (StanevPrime) [inline packages/rx_bloc_cli/mason_templates/bricks/rx_bloc_base/__brick__/README.md:293]: Same as above

**Review by StanevPrime — APPROVED** [review]
In general looks good. I left a couple of cosmetic comments. Please address them before you merge the PR

**Pull request merged**